addons/udes_security/models/models.py

```diff
@@ -7,9 +7,27 @@
 class BaseModel(models.AbstractModel):
     _inherit = "base"
 
+    @api.multi
+    def write(self, vals):
+        """Inherit write to add a check against res.archiving.restrictions
+        when the active field is written to.
+        """
+        ResArchivingRestrictions = self.env["res.archiving.restrictions"]
+        if "active" in vals:
+            ResArchivingRestrictions._check_can_archive(self._name)
+        return super().write(vals)
+
     @api.model
     def fields_view_get(self, view_id=None, view_type="form", toolbar=False, submenu=False):
-        """Override to remove buttons/editable functionality for view only desktop users"""
+        """Override to remove buttons/editable functionality for view only desktop users
+        and take into account archiving restrictions for the given user"""
+
+        def _remove_active(doc):
+            """Remove any field or smart button related to the active field from the doc"""
+            for node in doc.xpath("//button[@icon='fa-archive']"):
```

Comment:
We shouldn't be removing buttons based on what icon they have imo, unlikely be we could be unintentionally removing a button that is needed.

Could we change this to `@name='toggle_active'`? This is the core functionality for archiving in Odoo.